spring-boot/admin/src/main/resources/templates/admin/fragments/wizard/wizard.html

```diff
@@ -1,348 +1,389 @@
-
+        }
+    }
+
+    function validateServiceName(stepnumber) {
+        if ($("#serviceName").val().trim() == "") {
+            alert("Service name is mandatory");
+            return false;
+        }
+        return true;
+    }
+
+    function validateURL(stepnumber) {
+        if (stepnumber > 2) {
+            if ($("#serviceUrl").val().trim() == "") {
+                alert("Service URL is mandatory");
+                return false
+            } else if (!isValidURL($("#serviceUrl").val().trim())) {
+                alert("Service URL must be valid and start with http:// or https://");
+                return false;
             }
-
-            function onFinishCallback(objs, context) {
-
-                if ($("#serviceName").val().trim() == "")
-                {
-                    alert("You must fill service name");
-                }
-                if ($("#serviceUrl").val().trim() == "")
-                {
-                    alert("You must fill service url");
-                }
-
-                if (!isValidURL($("#serviceUrl").val().trim()))
-                {
-                    alert("Service URL must be valid and start with http:// or https://");
-                }
-
-                if ($("#serviceForFree").val() == "")
-                {
-                    alert("You must select either free service or not");
-                } else
-                {
-                    if ($("#serviceForFree").val() == "false" && $("#monthlyChargeInput").val() == "")
-                    {
-                        alert("You must select either monthly charge or charge per call");
-                    }
-
-                    if ($("#serviceForFree").val() == "false" && $("#serviceCost").val() == "")
-                    {
-                        alert("You must set up a cost for your service");
-                    }
-                }
-
-                if ($("#serviceName").val().trim() != "" && $("#serviceUrl").val().trim() && isValidURL($("#serviceUrl").val().trim()))
-                {
-                    if ($("#serviceForFree").val() !== "")
-                    {
-                        if ($("#serviceForFree").val() == "true")
-                        {
-                            $("#wizardForm").submit();
-                            $('#wizard').hide();
-                            $('#busy_indicator').show();
-                        } else
-                        {
-                            if ($("#monthlyChargeInput").val() !== "" && $("#serviceCost").val() !== "")
-                            {
-                                $("#wizardForm").submit();
-                                $('#wizard').hide();
-                                $('#busy_indicator').show();
-                            }
-                        }
-                    }
-
-                }
+        }
+        return true;
+    }
+
+    function validateServiceCharge(stepnumber) {
+        if (stepnumber > 3) {
```

Comment:
Same comment about `stepnumber` as for previous methods

Reply:
done